### DISCUSSION TITLE
X-Rite i1Profiler 3.7.0 breaks IM - conflict on system environment variable MAGICK_CODER_MODULE_PATH

### DISCUSSION THREAD

**Discussion by dtrout** [ImageMagick/ImageMagick#6759] [Development] [answered]
### ImageMagick version

7.1.1-19

### Operating system, version and so on

Windows 10

### Description

IM is broken after installing the latest version of X-Rite i1Profiler (3.7.0).  The problem manifests as IM not being able to find valid delegates for all file formats, e.g. JPG, TIFF, etc.

After a lot of debugging on my part, the problem is that installation of i1Profiler 3.7.0 instantiates a system environment variable MAGICK_CODER_MODULE_PATH which points to "C:\Program Files (x86)\X-Rite\i1Profiler\coders".  This directory has lots of dll's which do appear to be IM dll's, but I'm at a loss at why X-Rite is installing them for their product.  In any case, the presence of this environment variable causes IM to search in this directory for its delegates and it doesn't find valid delegate dll's.

This could be (is?) a bug on X-Rite's part, but I would ask the IM developers to contact them to get this straightened out.  It causes almost every conceivable IM command to fail due to not finding any delegates.  **X-Rite should NOT be creating this system  environment variable that is already in use by IM.**

**Comment by dlemstra** [comment, marked as answer]
We will not contact them about this. You will need to do that yourself.

**Reply by dtrout** [reply]
OK, done. At least you are now aware of the problem (and for me it is a BIG one).  Let's hope X-Rite will get this resolved quickly.

**Reply by dlemstra** [reply]
They are probably using our library in their product but they should really set `MAGICK_CODER_MODULE_PATH` differently. And make sure you point them to our discussions page if they need help with this.

**Comment by dtrout** [comment]
Update:  X-Rite has now released i1Profiler version 3.7.1 which fixes their bug.  Installing i1Profiler no longer adds MAGICK_CODER_MODULE_PATH to the system environment.

Note; If you previously installed i1Profiler 3.7.0, you may still have this environment variable in place, so I'd recommend you check your Windows System Environment variables.  If you aren't using MAGICK_CODER_MODULE_PATH to specify your own delegate dll's, then you should delete the variable (which i1Profiler 3.7.0 previously instantiated) as I don't think the i1Profiler 3.7.1 installer will remove it for you.
